api/scpca_portal/models/publication.py

```diff
@@ -20,3 +26,39 @@ def __str__(self) -> str:
     def doi_url(self):
         """Returns DOI URL."""
         return f"https://doi.org/{self.doi}"
+
+    @classmethod
+    def get_from_dict(cls, data: Dict):
+        publication = cls(
+            doi=data.get("doi"),
+            citation=data.get("citation"),
+            pi_name=data.get("pi_name"),
+        )
+
+        return publication
+
+    @classmethod
+    def bulk_create_from_project_data(cls, project_data, project):
+        """Creates a list of publication objects and saves them."""
+        publications = []
+
+        keys = ["citation_doi", "citation"]
+        for doi, citation in utils.get_csv_zipped_values(
+            project_data, *keys, model_name=cls.__name__
+        ):
+            if doi in common.IGNORED_INPUT_VALUES:
+                continue
+
+            # Skip if already in db
+            if Publication.objects.filter(doi=doi):
```

Comment:
I think this is the general flow we want.
```suggestion
            if existing_pub := Publication.objects.get(doi=doi).exists():
                if not project.publications.get(existing_pub).exists():
                   project.publications.add(existing_pub)
                 continue
```